File: mobiledevelopment/lib/src/config/currency.dart
import 'package:intl/intl.dart';

const currencyCode = 'PHP';
const currencySymbol = '₱';
const budgetLabel = 'Budget (₱)';

final _currencyFormat = NumberFormat.currency(
  locale: 'en_PH',
  symbol: currencySymbol,
  decimalDigits: 0,
);

final _currencyDecimalFormat = NumberFormat.currency(
  locale: 'en_PH',
  symbol: currencySymbol,
  decimalDigits: 2,
);

String formatCurrency(num amount) => _currencyFormat.format(amount);

String formatCurrencyDecimal(num amount) => _currencyDecimalFormat.format(amount);

File: src/config/currency.ts
export const CURRENCY_CODE = 'PHP';
export const CURRENCY_SYMBOL = '₱';
export const CURRENCY_LOCALE = 'en-PH';
export const BUDGET_LABEL = 'Budget (₱)';

export function formatCurrency(amount: number): string {
  return amount.toLocaleString(CURRENCY_LOCALE, {
    style: 'currency',
    currency: CURRENCY_CODE,
    minimumFractionDigits: 0,
    maximumFractionDigits: 0,
  });
}

export function formatCurrencyDecimal(amount: number): string {
  return amount.toLocaleString(CURRENCY_LOCALE, {
    style: 'currency',
    currency: CURRENCY_CODE,
    minimumFractionDigits: 2,
    maximumFractionDigits: 2,
  });
}
